tests/integration/suite/test_clustertemplate.py

```diff
@@ -46,7 +47,8 @@ def test_create_cluster_with_template(admin_mc, remove_resource):
         client.delete(cluster_template)
         assert e.value.error.status is None
 
-    remove_resource(cluster)
```

Comment:
We can still leave the remove_resource call immediately after cluster creation in case anything in `wait_for_cluster_to_be_deleted` fails.